### PR TITLE
add k8s labels to deployment service

### LINKED ISSUE #3137: Use Kubernetes recommended labels
We set a number of labels on the Camel K operator manifest itself, but also on the resources created by the operator.

We should aim at conforming to the Kubernetes recommended labels and deprecated the redundant ones: 

https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

### PR DESCRIPTION
**Pull request by desirekaleba**
Closes #3137 

**Release Note**
```release-note
NONE
```

### CODE REVIEW

File: config/manager/operator-deployment.yaml [hunk 1]
```diff
@@ -22,6 +22,10 @@ metadata:
   labels:
     app: "camel-k"
     camel.apache.org/component: operator
+    name: camel-k-operator
+    app.kubernetes.io/component: operator
+    app.kubernetes.io/name: camel-k
+    app.kubernetes.io/version: "1.8.2"
```
[the remark this answers is not in the document]
Comment (desirekaleba) [reply config/manager/operator-deployment.yaml:28]: @claudio4j, I am thinking about tweaking it with the following `"${VERSION}"`. Do I need to import anything to have access to Makefile variables?

[the remark this answers is not in the document]
Comment (claudio4j) [reply config/manager/operator-deployment.yaml:28]: Have a look at `script/set_version.sh`, probably you don't have to use kustomize for this case, just plain bash to replace the version value.

[the remark this answers is not in the document]
Comment (desirekaleba) [reply config/manager/operator-deployment.yaml:28]: @claudio4j, I have looked at `script/set_version.sh`, this is just setting the version which is expected as the first positional argument to `set_version.sh` execution and after echos `Camel K version set to: $version and image name to: $image_name`. I was thinking about how to update the version value in `app.kubernetes.io/version: "1.8.2"` every time `set_version.sh version_number` is executed but I don't see a way around that. Would you please provide some hints on this?

[the remark this answers is not in the document]
Comment (claudio4j) [reply config/manager/operator-deployment.yaml:28]: You would need to add the labels with fixed values as is in this PR, and whenever `set_version.sh` is invoked, it should change the `1.8.2` fixed value of the `app.kubernetes.io/version` label by using sed.

[the remark this answers is not in the document]
Comment (desirekaleba) [reply config/manager/operator-deployment.yaml:28]: @claudio4j, I will need your review on this, please.

**Review by claudio4j — APPROVED** [review]

**Review by astefanutti — APPROVED** [review]
Thanks! That'd be awesome if you could follow this up with adding the labels on the resources created by the operator itself.

**Pull request merged**